pubspec.yaml

```diff
@@ -3,9 +3,9 @@ name: pub
 dependencies:
   # Note: Pub's test infrastructure assumes that any dependencies used in tests
   # will be hosted dependencies.
-  analyzer: ">=0.25.0 <0.31.0"
+  analyzer: ^0.31.2-alpha.1
```

Comment:
Are these tighter constraints really necessary? I generally try to keep constraints matching the actual APIs that are being used and let the version solver sort out which versions of the package are compatible with my Dart VM.

Reply:
Would you prefer `">=0.31.2-alpha.1 <0.32.0"`?

Reply:
@kevmoo The same argument could be made for every version constraint here, but setting a policy of constantly revving them to match the SDK's versions is untenable. We have thorough integration tests in the SDK to make sure the versions work out, and otherwise we should treat this like any other package and only muck with version constraints when it's actually necessary.

Reply:
synced offline. I think that "works with Dart 2" is a new behavior for the package that we want to start depending on, and I think it makes sense for our pub constraint to reflect that.

In the end, as we discussed, for this package it's mostly academic.